### PR TITLE
Update TEPs template

### PR DESCRIPTION
**Pull request by jerop**
In this change, we update the TEPs template:
- Add questions that collaborators on a TEP should consider
to meet our design goals
- Reorganize the template to bring related sections together
- Fix typos and make other editorial changes

Co-authored-by: Christie Wilson <christiewilson@google.com>

cc @bobcatfish

### CODE REVIEW

File: teps/tools/tep-template.md.template [hunk 1]
```diff
@@ -127,17 +132,18 @@ and make progress.
 Describe the concrete improvement specific groups of users will see if the
 Motivations in this doc result in a fix or feature.
 
-Consider both the user's role (are they a Task author? Catalog Task user?
-Cluster Admin? etc...) and experience (what workflows or actions are enhanced
-if this problem is solved?).
+Consider the user's:
+- role - are they a Task author? Catalog Task user? Cluster Admin? e.t.c.
```
Comment (lbernick) [inline teps/tools/tep-template.md.template:136]: maybe link to our user profiles: https://github.com/tektoncd/community/blob/main/user-profiles.md

File: teps/tools/tep-template.md.template [hunk 2]
```diff
@@ -100,8 +104,8 @@ updates.
 
 <!--
 This section is for explicitly listing the motivation, goals and non-goals of
-this TEP.  Describe why the change is important and the benefits to users.  The
-motivation section can optionally provide links to [experience reports][] to
+this TEP. Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
```
Comment (lbernick) [inline teps/tools/tep-template.md.template:108]: I don't think this should have brackets

**Review by lbernick — COMMENTED** [review]
/lgtm

File: teps/tools/tep-template.md.template (hunk 2 above)
Comment (jerop) [reply teps/tools/tep-template.md.template]: added the link needed there 👍🏾

File: teps/tools/tep-template.md.template (hunk 1 above)
Comment (jerop) [reply teps/tools/tep-template.md.template]: added, thank you!

**Review by vdemeester — APPROVED** [review]

**Comment by bobcatfish** [comment]
Excellent!! :D Thanks @jerop 

/approve

**Comment by lbernick** [comment]
/lgtm

I think this should work now!

**Pull request merged**